src/main/java/com/yahoo/bullet/operations/aggregations/GroupBy.java

```diff
@@ -0,0 +1,114 @@
+package com.yahoo.bullet.operations.aggregations;
+
+import com.yahoo.bullet.BulletConfig;
+import com.yahoo.bullet.operations.aggregations.grouping.CachingGroupData;
+import com.yahoo.bullet.operations.aggregations.grouping.GroupData;
+import com.yahoo.bullet.operations.aggregations.grouping.GroupDataSummary;
+import com.yahoo.bullet.operations.aggregations.grouping.GroupOperation;
+import com.yahoo.bullet.operations.aggregations.sketches.TupleSketch;
+import com.yahoo.bullet.parsing.Aggregation;
+import com.yahoo.bullet.parsing.Specification;
+import com.yahoo.bullet.record.BulletRecord;
+import com.yahoo.bullet.result.Clip;
+import com.yahoo.bullet.result.Metadata.Concept;
+import com.yahoo.sketches.ResizeFactor;
+import com.yahoo.sketches.tuple.Sketch;
+import com.yahoo.sketches.tuple.SketchIterator;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * This {@link Strategy} implements a Tuple Sketch based approach to doing a group by. In particular, it
+ * provides a uniform sample of the groups if the number of unique groups exceed the Sketch size. Metrics like
+ * sum and count when summed across the uniform sample and divided the sketch theta gives an approximate estimate
+ * of the total sum and count across all the groups.
+ */
+public class GroupBy extends KMVStrategy<TupleSketch> {
+    private int size;
+    private final Map<String, String> fieldMapping;
```

Comment:
This is the map of fields to their new names?  What would you think about renaming this to "fieldsToNewNames" or something like that for clarity?  ...Similar to "fieldsToValues" in the consume() function?